code/control/GitDispatcher.php

```diff
@@ -81,17 +81,24 @@ public function update(SS_HTTPRequest $request) {
 	 * @return string
 	 */
 	public function show(\SS_HTTPRequest $request) {
+		$targetEnvironment = null;
+		$targetEnvironmentId = $request->getVar('environmentId');
+		if (!empty($targetEnvironmentId)) {
+			$targetEnvironment = DNEnvironment::get()->byId((int) $targetEnvironmentId);
+		}
+
 		$refs = [];
 		$prevDeploys = [];
 
 		$uatEnvironment = $this->project->DNEnvironmentList()->filter('Usage', 'UAT')->first();
 		$uatBuild = $uatEnvironment ? $uatEnvironment->CurrentBuild() : null;
-		if ($uatBuild && $uatBuild->exists()) {
+		if ($uatBuild && $uatBuild->exists() && $targetEnvironment && $targetEnvironment->Usage === 'Production') {
```

Comment:
We should probably change to use the `DNEnvironment::PRODUCTION` constant? I noticed that we also haven't used that in other places :(